backend/internal/controller/adapter/shot_message_test.go

```diff
@@ -0,0 +1,136 @@
+package adapter
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+
+	"github.com/HackYourCareer/SmartKickers/internal/config"
+	"github.com/HackYourCareer/SmartKickers/internal/model"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUnpackShotMsg(t *testing.T) {
+	type args struct {
+		name           string
+		msgIn          tableShotParams
+		expectedMsgOut model.Shot
+		expectedError  string
+	}
+	tests := []args{
+		{
+			name: "wrong area message, should return empty model.Shot and error",
+			msgIn: tableShotParams{
+				Speed:     25.5,
+				StartArea: 1,
+			},
+			expectedMsgOut: model.Shot{},
+			expectedError:  "couldn't decode teamID",
+		},
+		{
+			name: "should return speed 25.5 and team white",
+			msgIn: tableShotParams{
+				Speed:     25.5,
+				StartArea: 20,
+			},
+			expectedMsgOut: model.Shot{
+				Speed: 25.5,
+				Team:  config.TeamWhite,
+			},
+			expectedError: "",
+		},
+		{
+			name: "should return speed 17.226 and team blue",
+			msgIn: tableShotParams{
+				Speed:     17.226,
+				StartArea: 24,
+			},
+			expectedMsgOut: model.Shot{
+				Speed: 17.226,
+				Team:  config.TeamBlue,
+			},
+			expectedError: "",
+		},
+		{
+			name:           "no message, should return empty shot message and error",
+			msgIn:          tableShotParams{},
+			expectedMsgOut: model.Shot{},
+			expectedError:  "missing shot parameters",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var rawSlice []json.RawMessage
+			jsn, err := json.Marshal(tt.msgIn)
+			if err != nil {
+				assert.FailNow(t, err.Error())
+			}
+			rawMsg := json.RawMessage(jsn)
+			if tt.msgIn != (tableShotParams{}) {
+				rawSlice = append(rawSlice, rawMsg)
+			}
+
+			tableShot := tableShotMsg{
+				Params: rawSlice,
+			}
+			shotJSON, err := json.Marshal(tableShot)
+			if err != nil {
+				assert.FailNow(t, err.Error())
+			}
```

Comment:
Use `.Nil(t, err)` instead

Reply:
Is this a good idea? In the case of unsuccessful marshalling there is probably no point in performing further tests.